frontend/src/components/login.js

```diff
@@ -1,44 +1,165 @@
 import { useState } from 'react';
-import { loginUser } from '../http/user'
+import { registerUser, loginUser } from '../http/user'
 
-const Login = ({ setUserState }) => {
+const Login = ({ setUserState, setMessage, setReturnError }) => {
+    //login states
     const [userInput, setUserInput] = useState('');
     const [passwordInput, setPasswordInput] = useState('');
-    const [loginError, setLoginError] = useState('');
+    //register states
+    const [userRegInput, setUserRegInput] = useState('');
+    const [passwordRegInput, setPasswordRegInput] = useState('');
+    const [emailRegInput, setEmailRegInput] = useState('');
+    //error and toggler states
+    const [errorState, setErrorState] = useState('');
+    const [showRegister, setShowRegister] = useState(false)
 
 
+    //login function
     const handleLogin = event => {
         //trim username and set to lowercase
         const userTrim = userInput.trim();
         const passTrim = passwordInput.trim();
         event.preventDefault();
         loginUser(userTrim, passTrim).then(res => {
             if (res.name) {
+                setReturnError("");
                 setUserState(res.name);
             } else {
-                setLoginError(res);
+                setErrorState(res);
             }
         });
     }
-    return (
-        <div className="col-12 offset-md-4 col-md-4">
-            <form className="card mx-auto px-4 py-2">
-                <div className="card-body">
-                    <p className="card-title lead text-center mb-4">You must login to continue</p>
-                    <div className="input mb-3">
-                        <label className="input-text pb-2" htmlFor="username">Username</label>
-                        <input type="text" id="username" name="username" className="form-control" aria-label="username-input" aria-describedby="username-input" onChange={(e) => setUserInput(e.target.value)} />
+    //register function
+    const handleRegister = async (event) => {
+        event.preventDefault();
+        const res = await registerUser(userRegInput, emailRegInput, passwordRegInput);
+        if (!res.name) {
+            setErrorState(res.error);
+            return res.error;
+        } else {
+            setReturnError("")
+            setMessage('Successfully created account.')
+            setUserRegInput('');
+            setPasswordRegInput('');
+            setEmailRegInput('');
+            setTimeout(() => {
+                setMessage('')
```

Comment:
**No need to change** but: this is what I meant by having FlashMessage handle timeout. This type of logic does not belong here IMO, it is the responsibility of the FlashMessage component to... flash the message. Not another component.

What happens when johnny appleseed comes along and adds another component that needs to flash a message and then sets the timeout for 10000 instead of 1000 😃

Reply:
Johnny appleseed will have to Johnny applerub deez applenuts! Idk how to set the timeout properly in the flashmessage component. Will have a look at it after i finish this confirmation password stuff.